src/__tests__/type.js

```diff
@@ -1022,3 +1023,83 @@ test('should not type inside a contenteditable=false div', () => {
+  userEvent.type(element, 'Brea{arrowleft}k{arrowright}fast')
+  expect(getEventSnapshot()).toMatchInlineSnapshot(`
+    Events fired on: input[value="Brekafast"]
+
+    input[value=""] - pointerover
+    input[value=""] - pointerenter
+    input[value=""] - mouseover: Left (0)
+    input[value=""] - mouseenter: Left (0)
+    input[value=""] - pointermove
+    input[value=""] - mousemove: Left (0)
+    input[value=""] - pointerdown
+    input[value=""] - mousedown: Left (0)
+    input[value=""] - focus
+    input[value=""] - focusin
+    input[value=""] - pointerup
+    input[value=""] - mouseup: Left (0)
+    input[value=""] - click: Left (0)
+    input[value=""] - keydown: B (66)
+    input[value=""] - keypress: B (66)
+    input[value="B"] - input
+      "{CURSOR}" -> "B{CURSOR}"
+    input[value="B"] - keyup: B (66)
+    input[value="B"] - keydown: r (114)
+    input[value="B"] - keypress: r (114)
+    input[value="Br"] - input
+      "B{CURSOR}" -> "Br{CURSOR}"
+    input[value="Br"] - keyup: r (114)
+    input[value="Br"] - keydown: e (101)
+    input[value="Br"] - keypress: e (101)
+    input[value="Bre"] - input
+      "Br{CURSOR}" -> "Bre{CURSOR}"
+    input[value="Bre"] - keyup: e (101)
+    input[value="Bre"] - keydown: a (97)
+    input[value="Bre"] - keypress: a (97)
+    input[value="Brea"] - input
+      "Bre{CURSOR}" -> "Brea{CURSOR}"
+    input[value="Brea"] - keyup: a (97)
+    input[value="Brea"] - keydown: ArrowLeft (37)
+    input[value="Brea"] - select
+    input[value="Brea"] - keyup: ArrowLeft (37)
+    input[value="Brea"] - keydown: k (107)
+    input[value="Brea"] - keypress: k (107)
+    input[value="Breka"] - input
+      "Bre{CURSOR}a" -> "Breka{CURSOR}"
+    input[value="Breka"] - select
+    input[value="Breka"] - keyup: k (107)
+    input[value="Breka"] - keydown: ArrowRight (39)
+    input[value="Breka"] - select
+    input[value="Breka"] - keyup: ArrowRight (39)
+    input[value="Breka"] - keydown: f (102)
+    input[value="Breka"] - keypress: f (102)
+    input[value="Brekaf"] - input
+      "Breka{CURSOR}" -> "Brekaf{CURSOR}"
+    input[value="Brekaf"] - keyup: f (102)
+    input[value="Brekaf"] - keydown: a (97)
+    input[value="Brekaf"] - keypress: a (97)
+    input[value="Brekafa"] - input
+      "Brekaf{CURSOR}" -> "Brekafa{CURSOR}"
+    input[value="Brekafa"] - keyup: a (97)
+    input[value="Brekafa"] - keydown: s (115)
+    input[value="Brekafa"] - keypress: s (115)
+    input[value="Brekafas"] - input
+      "Brekafa{CURSOR}" -> "Brekafas{CURSOR}"
+    input[value="Brekafas"] - keyup: s (115)
+    input[value="Brekafas"] - keydown: t (116)
+    input[value="Brekafas"] - keypress: t (116)
+    input[value="Brekafast"] - input
+      "Brekafas{CURSOR}" -> "Brekafast{CURSOR}"
+    input[value="Brekafast"] - keyup: t (116)
+  `)
+})
+
+test('navigationKey throws if an invalid key is passed', () => {
+  expect(() => {
+    navigationKey('SOME_INVALID_KEY')
+  }).toThrowError(TypeError)
+})
```

Comment:
I've modified the function to hit all the lines (`default` wasn't reachable) now the question is, should I include this test to make sure we throw a `TypeError` (and also throw it in the function)` if we don't get a valid key, or leave it without throwing, since it's an internal function anyway.

In summary, removing the last commit entirely, we still get 💯 coverage, but it's not really covered if an invalid key is passed, any thoughts? @juanca @kentcdodds

Reply:
Note, modified the commit because I didn't have my git config setup properly in this  machine, but yeah, same changes 😅

Reply:
Is this even possible? We're the only ones who call `navigationKey` and we always call it properly. I'd just remove that logic.

Reply:
No, it isn't I'll remove it :)